packages/compass-e2e-tests/index.js

```diff
@@ -35,11 +39,36 @@ async function main() {
     mocha.addFile(path.join(__dirname, testPath));
   });
 
-  mocha.run((failures) => (process.exitCode = failures ? 1 : 0));
+  keychain.activate();
+
+  mocha.run((failures) => {
+    keychain.reset();
+    process.exitCode = failures ? 1 : 0;
+  });
 }
 
+process.on('SIGINT', () => {
+  debug(`Process was interrupted. Resetting keychain to default and exiting.`);
+  keychain.reset();
+  process.exitCode = 128 + SIGINT;
+  process.exit();
+});
+
+process.on('SIGTERM', () => {
+  debug(`Process was terminated. Resetting keychain to default and exiting.`);
+  keychain.reset();
+  process.exitCode = 128 + SIGTERM;
+  process.exit();
+});
+
+process.on('uncaughtException', () => {
+  debug('Uncaught exception. Resetting keychain to default and exiting.');
+  keychain.reset();
+});
```

Comment:
@addaleax does doing something like this makes sense to you?

```suggestion
process.once('uncaughtException', (err) => {
  debug('Uncaught exception. Resetting keychain to default and exiting.');
  keychain.reset();
  throw err;
});
```

... or should I do what Node.js docs are saying and do a `console.error(err); process.exitCode = 1; process.exit();` instead?

Reply:
Honestly, I think either way is just fine :)

I think I would personally also do what you’re doing – it has the implication that the exit code will be 7 instead of 1, but honestly, in the end that doesn’t matter all that much :)

Reply:
Alright, will go with this option then, thanks!